src/PhpParser/Node/BetterNodeFinder.php

```diff
@@ -78,17 +77,14 @@ public function findParentType(Node $node, string $type): ?Node
         Assert::isAOf($type, Node::class);
 
         $parentNode = $node->getAttribute(AttributeKey::PARENT_NODE);
-        if (! $parentNode instanceof Node) {
-            return null;
-        }
 
-        do {
+        while ($parentNode instanceof Node) {
             if (is_a($parentNode, $type, true)) {
```

Comment:
I guess it could be faster to use the language construct `instanceof $type` instead of `is_a`

Reply:
Ok, updated with `instanceof` https://github.com/rectorphp/rector-src/pull/3258/commits/ec88c2e7a918d81098dba8dea52ec11969f7968e

Reply:
updated with MultiInstanceofChecker service with update pass only object https://github.com/rectorphp/rector-src/pull/3258/commits/3d4ff4a0405062271d3794bb68b6fcbec6fa149e

Reply:
Using `MultiInstanceofChecker` service seems cause phpstan notice:

```
composer phpstan
> vendor/bin/phpstan analyse --ansi --error-format symplify

 ! [NOTE] The Xdebug PHP extension is active, but "--xdebug" is not used. This may slow down performance and the process
 !        will not halt at breakpoints.                                                                                 

Note: Using configuration file /home/abdul/www/rector-src/phpstan.neon.
 2511/2511 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%

 ------------------------------------------------------------------------------------------------------------------------------------------------------------
  src/PhpParser/Node/BetterNodeFinder.php:64
 ------------------------------------------------------------------------------------------------------------------------------------------------------------
  - '#Method Rector\\Core\\PhpParser\\Node\\BetterNodeFinder\:\:findParentByTypes\(\) should return \(TNode of PhpParser\\Node\)\|null but returns PhpParser\\Node#'
 ------------------------------------------------------------------------------------------------------------------------------------------------------------
```

Reverting https://github.com/rectorphp/rector-src/pull/3258/commits/facf8f4b39836734a6fbe6b684eaaaa26a9179cf